tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/LegacyValidOnContainerCascadingTest.java

```diff
@@ -0,0 +1,304 @@
+/**
+ * Bean Validation TCK
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.beanvalidation.tck.tests.validation.graphnavigation.containerelement;
+
+import static org.hibernate.beanvalidation.tck.util.ConstraintViolationAssert.assertThat;
+import static org.hibernate.beanvalidation.tck.util.ConstraintViolationAssert.pathWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Valid;
+import javax.validation.Validator;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.beanvalidation.tck.beanvalidation.Sections;
+import org.hibernate.beanvalidation.tck.tests.AbstractTCKTest;
+import org.hibernate.beanvalidation.tck.tests.validation.graphnavigation.containerelement.model.ExtendedChecks1;
+import org.hibernate.beanvalidation.tck.tests.validation.graphnavigation.containerelement.model.MyListWithGroupConversions;
+import org.hibernate.beanvalidation.tck.tests.validation.graphnavigation.containerelement.model.ValidOnListAndOnTypeArgumentWithGroupConversions;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecAssertion;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+/**
+ * Test that the legacy behavior of having @Valid on the container works correctly: it validates the elements of the
+ * iterable AND the properties of the container if any.
```

Comment:
I would prefer to _not_ test that latter part in the TCK. The reason being that we don't clearly state that it should happen:

> [@Valid] causes the contents of the iterator to be validated [...] Each object provided by the iterator is validated. For Map, the value (retrieved by getValue) of each Map.Entry is validated (the key is not validated).

So what HV is doing, goes beyond the spec, but I don't think it can be considered portable behavior and thus can be tested in the TCK.